src/client/sandbox/cookie/server-sync.js

```diff
@@ -0,0 +1,122 @@
+import nativeMethods from '../../sandbox/native-methods';
+import XhrSandbox from '../../sandbox/xhr';
+import settings from '../../settings';
+import { stringify as stringifyJSON } from '../../json';
+import { isIE9 } from '../../utils/browser';
+
+const SWITCH_BACK_TO_ASYNC_XHR_DELAY = 2000;
+
+export default class ServerSync {
+    constructor () {
+        this.useAsyncXhr = true;
+        this.queue       = [];
+        this.sendedQueue = null;
+        this.activeReq   = null;
+
+        // NOTE: When unloading, we should switch to synchronous XHR to be sure that we won't lose any cookies.
+        nativeMethods.windowAddEventListener.call(window, 'beforeunload', () => this._beforeUnloadHandler(), true);
+    }
+
+    static _createXMLHttpRequest (isAsync) {
+        var xhr = XhrSandbox.createNativeXHR();
+
+        xhr.open('POST', settings.get().cookieSyncUrl, isAsync);
```

Comment:
![image](https://cloud.githubusercontent.com/assets/5373460/26357056/c7e5d910-3fd6-11e7-8e10-0a35376a7ebb.png)
In some code editors we have wrong highlight because `async` is a keyword in the es7 standard.